src/modal/modal.js

```diff
@@ -129,6 +151,33 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.stackedMap'])
         element.addClass(attrs.windowTopClass || '');
         scope.size = attrs.size;
 
+        // get topmost modal object
+        var bodyIsOverflowing;
+        var modal = $modalStack.getTop();
+        
+        // only when modal is attached to body
+        if(modal && modal.value && modal.value.appendTo === 'body'){
+          
+          // check bodyOverflowing property that was set when opening modal
+          if(modal.value.modalDomEl && modal.value.modalDomEl.bodyOverflowing){
+            bodyIsOverflowing = true;
+          }
+
+          // start - from adjustDialog method of modal.js of bootstrap
+          // check if modal is overflowing
+          var modalIsOverflowing = element[0].scrollHeight > document.documentElement.clientHeight;
+          
+          if(!$$scrollbarHelper.scrollbarWidth){
+            $$scrollbarHelper.measureScrollbar();
+          }
+
+          element.css({
+            'padding-left': (!bodyIsOverflowing && modalIsOverflowing ? $$scrollbarHelper.scrollbarWidth : '') + 'px',
+            'padding-right': (bodyIsOverflowing && !modalIsOverflowing ? $$scrollbarHelper.scrollbarWidth : '') + 'px'
```

Comment:
This means that the object could potentially be setting
```js
element.css({
  'padding-left': 'px',
  'padding-right': 'px'
});
```
It should be setting 0 at worse.